1-source-code/2-writing/2-gofmt-and-friends.md

```diff
@@ -302,6 +302,59 @@ type Person struct {
 }
 ```
 
+## `eg`
+
+`eg` is an example-based refactoring tool.
+To use it, you simply create a template file and then provide the paths you wish to refactor.
+`eg` is particulary useful for changing function or API usages.
+
+### Installing `eg`
+
+```bash
+$ go get golang.org/x/tools/cmd/eg
+$ ^get^build
+```
+
+### Writing a template
+
+Here's a simple template (`template.go`) to get you started.
+Templates need to be buildable Go files with `before` and `after` functions.
+
+```go
+package template
+
+import (
+	"errors"
+	"fmt"
+)
+
+func before(s string) error { return fmt.Errorf("%s", s) }
+func after(s string) error  { return errors.New(s) }
+```
+
+### Running `eg`
+
+```bash
```

Comment:
Separate this into some Go code using embedmd, then explain that running that command (in a bash block) will change the file into something else

Reply:
Good idea -- done.